pkg/signature/kms/azure/client_test.go

```diff
@@ -235,3 +235,58 @@ func testSetEnv(t *testing.T, s map[string]string) func() {
 		}
 	}
 }
+
+func TestParseReference(t *testing.T) {
```

Comment:
The content from `azure_test.go` was copied here. Since all of the tests in that file were around testing the client, it made sense to move them here.